File: service/app/src/main/java/io/miragon/blueprint/application/port/inbound/OrderBikeUseCase.java
package io.miragon.blueprint.application.port.inbound;

import io.miragon.blueprint.domain.bike.OrderId;
import io.miragon.blueprint.domain.leasing.ApplicationId;

public interface OrderBikeUseCase {

    Result orderBike(ApplicationId id);

    /** {@code orderId} is {@code null} when the requested bike was out of stock and no order was placed. */
    record Result(OrderId orderId, boolean bikeAvailable) {
    }
}

File: service/app/src/main/kotlin/io/miragon/blueprint/application/port/inbound/OrderBikeUseCase.kt
package io.miragon.blueprint.application.port.inbound

import io.miragon.blueprint.domain.leasing.ApplicationId
import io.miragon.blueprint.domain.bike.OrderId

interface OrderBikeUseCase {
    fun orderBike(id: ApplicationId): Result

    data class Result(
        // null when the requested bike was out of stock and no order was placed
        val orderId: OrderId?,
        val bikeAvailable: Boolean,
    )
}
